core/src/main/java/org/apache/carbondata/core/scan/executor/util/RestructureUtil.java

```diff
@@ -200,8 +207,14 @@ private static boolean isColumnMatchesStruct(CarbonColumn tableColumn, CarbonCol
         }
         carbonDimension = CarbonTable.getCarbonDimension(tempColName.toString(), parentDimension);
         if (carbonDimension != null) {
+          // In case of SDK the columnId and columnName is same and this check will ensure for
+          // all the child columns that the table column name is equal to query column name and
+          // table columnId is equal to table columnName
           if (carbonDimension.getColumnSchema().getColumnUniqueId()
-              .equalsIgnoreCase(queryColumn.getColumnId())) {
+              .equalsIgnoreCase(queryColumn.getColumnId()) || (
+              carbonDimension.getColumnSchema().getColumnUniqueId()
```

Comment:
Why there is a diff of condition check for dimension and measure? Here we compare with ColumnUniqueId but other places are not

Reply:
This is only for the case of struct matching, as was before, just added one more check.